frontend/app/component/form/polling_station/PollingStationForm.test.tsx

```diff
@@ -34,32 +34,116 @@ async function fillForm(user: UserEvent, testPollingStation: PollingStation | Om
+
+      await userEvent.click(screen.getByRole("button", { name: "Opslaan en toevoegen" }));
+
+      await waitFor(() => {
+        expect(onSaved).toHaveBeenCalledWith(expect.objectContaining(testPollingStation));
+      });
+    });
 
-    await waitFor(() => {
-      expect(onSaved).toHaveBeenCalledWith(expect.objectContaining(testPollingStation));
+    test("Validation", async () => {
+      const testObj: Omit<PollingStation, "id"> = {
+        election_id: 1,
+        number: 42,
+        name: "test",
+        street: "test",
+        postal_code: "1234",
+        locality: "test",
+        polling_station_type: "FixedLocation",
+        number_of_voters: 1,
+        house_number: "test",
+      };
+
+      const onSaved = vi.fn();
+      render(<PollingStationForm electionId={1} onSaved={onSaved} />);
+
+      const user = userEvent.setup();
+      await user.click(screen.getByRole("button", { name: "Opslaan en toevoegen" }));
+
+      const numberInput = screen.getByRole("textbox", { name: "Nummer" });
+      const nameInput = screen.getByRole("textbox", { name: "Naam" });
+      const numberOfVotersInput = screen.getByRole("textbox", { name: "Aantal kiesgerechtigden Optioneel" });
+      const streetInput = screen.getByRole("textbox", { name: "Straatnaam" });
+      const houseNumberInput = screen.getByRole("textbox", { name: "Huisnummer" });
+      const houseNumberAdditionInput = screen.getByRole("textbox", { name: "Toevoeging" });
+      const postalCodeInput = screen.getByRole("textbox", { name: "Postcode" });
+      const localityInput = screen.getByRole("textbox", { name: "Plaats" });
+      const typeInputOption = screen.getByRole("radio", { name: "Vaste locatie" });
+
+      await user.click(typeInputOption);
+
+      await waitFor(() => {
+        expect(numberInput).toBeInvalid();
+        expect(nameInput).toBeInvalid();
+      });
+
+      await user.type(nameInput, testObj.name);
+      await user.type(streetInput, testObj.street);
+      await user.type(houseNumberInput, testObj.house_number);
+      await user.type(postalCodeInput, testObj.postal_code);
+      await user.type(localityInput, testObj.locality);
+      await user.type(numberOfVotersInput, testObj.number_of_voters?.toString() || "");
+
+      await user.click(screen.getByRole("button", { name: "Opslaan en toevoegen" }));
+
+      await waitFor(() => {
+        expect(numberInput).toHaveAttribute("aria-invalid", "true");
+        expect(nameInput).toHaveAttribute("aria-invalid", "false");
+        expect(streetInput).toHaveAttribute("aria-invalid", "false");
+        expect(houseNumberInput).toHaveAttribute("aria-invalid", "false");
+        expect(postalCodeInput).toHaveAttribute("aria-invalid", "false");
+        expect(localityInput).toHaveAttribute("aria-invalid", "false");
+        expect(numberOfVotersInput).toHaveAttribute("aria-invalid", "false");
+        expect(houseNumberAdditionInput).toHaveAttribute("aria-invalid", "false");
```

Comment:
There's [`toBeInvalid()`](https://github.com/testing-library/jest-dom?tab=readme-ov-file#tobeinvalid) for this.

Reply:
changed